example/ffi/ndt.cpp

```diff
@@ -15,15 +15,15 @@ int main() {
     mk_task_t *task = nullptr;
     mk_task_error_t err =
             mk_task_start_ex(&task, "{\n"
```

Comment:
The reason why I don't use raw literals in that file is that I am using the `ffi` API and ideally I would even want to write that file in C to show you can really get away with a simple API. What prevents me from doing so is that I cannot force statically linking libc++ on macOS. Otherwise, I would really want to have that show case to proof that it is possible.

> But maybe there are object lifetime issues preventing this approach?

I think I am lacking some knowledge about raw literals and life cycles so that I do not understand very well what you mean here. Hence, please could you:

![supreme_dalek_gif__by_sallyfriceiii-d7qozkq](https://user-images.githubusercontent.com/337298/37846314-8f3c8316-2ecd-11e8-9562-360612cea76a.gif)

?

Reply:
`char *` constant typically live in some static allocation. `std::str` variables may or may not. So, if you were to save the passed-in `char *` without copying the string, and it came from a `std::str("fsdfds").c_str()` that was only allocated for that one call, then later use of that pointer could cause problems.

Reply:
Ah, right! So, since the pointer is used _immediately_ and not stored or use later, I believe C++ guarantees that the temporary `std::str` would be alive for long enough. (Still, as I mentioned above, I'd like that file to be written in C++ close to C as much as possible.)